collector/accelerators.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type acceleratorsCollector struct {
+	pciDevicesPath string
+	logger         log.Logger
+}
+
+func init() {
+	registerCollector("accelerators", defaultEnabled, NewAcceleratorCollector)
+}
+
+// NewXfrmCollector returns a new Collector exposing XFRM stats.
+func NewAcceleratorCollector(logger log.Logger) (Collector, error) {
+	return &acceleratorsCollector{
+		pciDevicesPath: filepath.Join(*sysPath, "bus/pci/devices"),
+		logger:         logger,
+	}, nil
+}
+
+var (
+	acceleratorCardsDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "accelerator", "cards_count"),
+		"number of accelerator cards",
+		[]string{"type"}, nil,
```

Comment:
should we have 2 labels, `vendor` and `type` (or `model`)?

Reply:
eventually we want to publish metrics that says, for example: 
"NVIDIA R600" : 2,
"QUALCOMM A100": 3
etc'.  
adding vendor label won't mean much, since nobody remembers what is the vendor number of NVIDIA, AMD or QUALCOMM

Reply:
My suggestion was more about exposing 2 labels so we can slice/dice the data more easily:
````
node_accelarator_cards{model="NVIDIA", type="A100"} 1
node_accelarator_cards{model="NVIDIA", type="V100"} 2
````

The advantage is that it will help when aggregating data on the Telemetry side ("how many NVIDIA cards are deployed overall?" is easier to answer).

Reply:
ok, i see what you mean. Yes, it seems better, i will change the code